src/pip/_internal/commands/freeze.py

```diff
@@ -74,8 +71,13 @@ def add_options(self):
     def run(self, options, args):
         # type: (Values, List[str]) -> int
         skip = set(stdlib_pkgs)
-        if not options.freeze_all:
-            skip.update(DEV_PKGS)
+
+        if options.freeze_all_used:
+            deprecated(
+                "--all is now the default behavior and does nothing",
+                None,
+                gone_in="23.3",
```

Comment:
```suggestion
                gone_in="23.0",
```

Let's keep this for as little time as possible?